src/transaction.js

```diff
@@ -93,15 +127,23 @@ Transaction.prototype.isCoinbase = function () {
   return this.ins.length === 1 && Transaction.isCoinbaseHash(this.ins[0].hash)
 }
 
-var EMPTY_SCRIPT = new Buffer(0)
-
 Transaction.prototype.addInput = function (hash, index, sequence, scriptSig) {
+  return this._addInput(hash, index, sequence, scriptSig, undefined)
+}
+
+Transaction.prototype.addWitnessInput = function (hash, index, sequence, witness) {
+  return this._addInput(hash, index, sequence, undefined, witness)
+}
```

Comment:
I think we can safely say we'll almost exclusively see P2SH wrapped segwit because 
1) there's no address for non P2SH segwit
2) it will result in addresses and scripts that can be send to by old nodes